task2/script.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+function isArrayOne(arr) {
+  return arr.constructor === Array;
+}
+
+function isArrayTwo(arr) {
+  return arr instanceof Array;
+}
+
+function isArrayThree(arr) {
+  return {}.toString.call(arr) === '[object Array]'
+}
+
+function range() {
+  var arr = [];
+  switch (arguments.length) {
+    case 1:
+      {
+        for (var i = 0; i < arguments[0]; i++) {
+          arr.push(i);
+        }
+      }
+      break;
+    case 2:
+      {
+        for (var i = arguments[0]; i < arguments[1]; i++) {
+          arr.push(i);
+        }
+      }
+      break;
+    case 3:
+      {
+        for (var i = arguments[0]; i < arguments[1]; i++) {
+          if (i % (arguments[2]) !== 0) {
+            arr.push(i)
+          }
+        }
+      }
+      break;
+  }
+  return arr;
+}
+
+function compact(arr) {
+  return arr.filter(function(arr) {
+    return Boolean(arr).valueOf() === true;
```

Comment:
looks like I see the same second time
please read about filter, falsy values, type converting